core/templates/dev/head/pages/exploration_editor/editor_tab/StateEditor.js

```diff
@@ -168,8 +170,17 @@ oppia.factory('trainingModalService', [
                   editorContextService.getActiveStateName();
                 var state = explorationStatesService.getState(currentStateName);
 
+                // Retrieve the interaction ID.
+                var interactionId = stateInteractionIdService.savedMemento;
+
+                var rulesServiceName = interactionId.charAt(0).toLowerCase() +
+                interactionId.slice(1) + 'RulesService';
```

Comment:
A new service would be good, I like that. I don't know where the best place to create it is, though; feel free to find somewhere suitable in the directory that's the LCA of the files that need it.